test/ctest_helpers.cmake

```diff
@@ -37,6 +37,29 @@ function(create_test_executable target)
 
     # Define our test executable.
     add_executable(${target} ${CREATE_TEST_SOURCES})
+
+    # For MacOS builds we need to generate a dSYM bundle that contains the debug symbols for each 
+    # executable.
+    if (WT_DARWIN AND HAVE_DIAGNOSTIC)
+        if ("${CREATE_TEST_EXECUTABLE_NAME}" STREQUAL "")
+            add_custom_command(
+                TARGET ${target} POST_BUILD
+                COMMAND dsymutil ${target}
+                WORKING_DIRECTORY ${test_binary_dir}
+                COMMENT "Running dsymutil on ${target}"
+                VERBATIM
+            )
+        else()
+            add_custom_command(
+                TARGET ${target} POST_BUILD
+                COMMAND dsymutil ${CREATE_TEST_EXECUTABLE_NAME}
+                WORKING_DIRECTORY ${test_binary_dir}
```

Comment:
I understood this when you explained it to me, but reading the code - there's a lot of duplication here, is there a downside to using `${target}` for both?

Reply:
I'll make a temporary variable and set it to either `target` or `CREATE_TEST_EXECUTABLE_NAME` then use that temporary variable in the `add_custom_command`

Reply:
That works too, but my question is more about whether we can just use `${target}` for both.

Reply:
The problem is we need to use the name of the executable file that's eventually created. So just below the `add_custom_command` we set the `OUTPUT_NAME` property of the target executable which renames the executable file in the end. If it's possible to access that property as a variable in the `add_custom_command` we could move the command after we set that property and just the OUTPUT_NAME as the variable. I'm not sure if that's possible though.